stl/inc/xstring

```diff
@@ -278,13 +343,22 @@ struct char_traits<unsigned short> : _WChar_traits<unsigned short> {};
 
 // STRUCT TEMPLATE _Narrow_char_traits
 template <class _Elem, class _Int_type>
-struct _Narrow_char_traits { // Implement char_traits for narrow character types char and char8_t
+struct _Narrow_char_traits : private _Char_traits<_Elem, _Int_type> {
+    // Implement char_traits for narrow character types char and char8_t
+private:
+    using _Primary_char_traits = _Char_traits<_Elem, _Int_type>;
+
+public:
     using char_type  = _Elem;
     using int_type   = _Int_type;
     using pos_type   = streampos;
     using off_type   = streamoff;
     using state_type = mbstate_t;
 
+    using _Primary_char_traits::_Copy_s;
+    using _Primary_char_traits::copy;
+    using _Primary_char_traits::move;
+
     _NODISCARD static _CONSTEXPR17 int compare(_In_reads_(_Count) const _Elem* const _First1,
         _In_reads_(_Count) const _Elem* const _First2, const size_t _Count) noexcept /* strengthened */ {
         // compare [_First1, _First1 + _Count) with [_First2, ...)

```

Comment:
Why isn't the body of this function simply `return __builtin_memcmp(_First1, _First2, _Count);`? I remember deciding that we should use `__builtin_u8memcmp` because we could, not because it actually made any difference. I also think I recall that we were reticent to simply use `__builtin_memcp` all the time out of fear of regressing default mode with bugs in the shiny new C++17 intrinsic. But looking at this mess today, `return __builtin_memcmp(_First1, _First2, _Count);` is really attractive.

Reply:
(Feel free to file an issue asking me to investigate this and the following comment - I recognize that we've let these `char_traits` changes creep beyond the scope of the PR, and I'm asking for more.)

Reply:
Will submit a followup PR.